src/faldbt/project.py

```diff
@@ -171,48 +194,93 @@ def get_scripts(self, keyword: str, before: bool) -> List[str]:
             return scripts_node.get("after") or []
 
 
+@dataclass
+class DbtRunResult:
+    nativeRunResult: Optional[RunResultsArtifact]
+
+    @property
+    def results(self) -> Sequence[RunResultOutput]:
+        if self.nativeRunResult:
+            return self.nativeRunResult.results
+        else:
+            return []
+
+
+@dataclass
+class DbtFreshnessExecutionResult:
+    _artifact: Optional[FreshnessExecutionResultArtifact]
+
+    @property
+    def results(self) -> Sequence[FreshnessNodeOutput]:
+        if self._artifact:
+            return self._artifact.results
+        else:
+            return []
+
+
 @dataclass
 class DbtManifest:
     nativeManifest: Manifest
 
-    def get_models(self) -> List[DbtModel]:
-        return list(
-            map(
-                lambda node: DbtModel(node),
-                filter(
-                    lambda node: node.resource_type == NodeType.Model,
-                    self.nativeManifest.nodes.values(),
-                ),
-            )
+    def get_model_nodes(self) -> Iterable[ManifestNode]:
+        return (
+            node
+            for node in self.nativeManifest.nodes.values()
+            if node.resource_type == NodeType.Model
         )
 
-    def get_tests(self) -> List[DbtTest]:
-        return list(
-            map(
-                lambda node: DbtTest(node),
-                filter(
-                    lambda node: node.resource_type == NodeType.Test,
-                    self.nativeManifest.nodes.values(),
-                ),
-            )
+    def get_test_nodes(self) -> Iterable[ManifestNode]:
+        return (
+            node
+            for node in self.nativeManifest.nodes.values()
+            if node.resource_type == NodeType.Test
         )
 
-    def get_sources(self) -> List[ParsedSourceDefinition]:
-        return list(
-            map(lambda node: DbtSource(node=node), self.nativeManifest.sources.values())
-        )
+    def get_source_nodes(self) -> Iterable[ParsedSourceDefinition]:
+        return self.nativeManifest.sources.values()
 
+    def _map_nodes(
+        self,
+        run_results: DbtRunResult,
+        freshness_results: DbtFreshnessExecutionResult,
+        generated_models: Dict[str, Path],
+    ) -> Tuple[List[DbtModel], List[DbtSource], List[DbtTest]]:
+        status_map = dict((r.unique_id, r.status) for r in run_results.results)
```

Comment:
nit: maybe let's start using dict comprehensions
```suggestion
        status_map = {r.unique_id: r.status for r in run_results.results}
 ```